core/src/main/java/org/apache/iceberg/util/NumberUtil.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.util;
+
+public class NumberUtil {
+
+  private NumberUtil() {}
+
+  /**
+   * @param value the string to convert, can be null
+   * @return parsed integer, returns null if the string is null
+   */
+  public static Integer createInteger(String value) {
```

Comment:
I don't think we need this class. We are reading `current.properties().get(TableProperties.AVRO_COMPRESSION_LEVEL)` as a String in order to convert it to an Integer, just to convert it later back to a String

Reply:
We don't need this class. Instead of `NumberUtil.createInteger` we can use `PropertyUtil.propertyAsNullableInt` on the property `Map` and property name, and that gets the value of the property as an Integer.